coral/src/app/features/requests/acls/AclRequests.tsx

```diff
@@ -68,11 +77,55 @@ function AclRequests() {
     keepPreviousData: true,
   });
 
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const {
+    isLoading: deleteIsLoading,
+    mutate: deleteRequest,
+    isSuccess: deleteIsSuccess,
+  } = useMutation({
+    mutationFn: deleteAclRequest,
+    onSuccess: (responses) => {
+      queryClient.refetchQueries(["aclRequests"]);
```

Comment:
I just noticed that while implementing deleting for schema requests. 
We should not do the refetch here I think?

```      
  onSuccess: (responses) => {
     
      // 🤔 refetching here means that we're calling the api without knowing 
      // if our deletion was successfull, so in case of an error, it's an unnecessary call
       queryClient.refetchQueries(["aclRequests"]);
        const response = responses[0];

        if (response.result !== "success") {

          setErrorQuickActions(
            response.message || response.result || "Unexpected error"
          );
        } else {
        // 🤔  only here we need to actually refetch, because only now we
        //  will have a different response from the api
          setErrorQuickActions("");
          
         // 🤔 here we're checking if the recently deleted 
        // request was the last on the currentPage, we still do that
          if (
            schemaRequests?.entries.length === 1 &&
            schemaRequests?.currentPage > 1
          ) {
           // 💡 this will actually trigger the api call for requests anyway, so we 
          // could add a else here
            return setCurrentPage(schemaRequests?.currentPage - 1);
          } else {          
          // 🤔 only here we need to refetch the query manually
              queryClient.refetchQueries(["getSchemaRequests"]);
           }
        }
      },
```


WDYT? 

(maybe my brain is also 🍝  at this point, but I feel that should avoid unnecessary calls?)

Reply:
That first refetch is for sure completely superfluous! I will remove it. But looking into this more closely actually made me uncover a bug we had in Approvals :o https://github.com/aiven/klaw/pull/878

Reply:
[77e3b86](https://github.com/aiven/klaw/pull/876/commits/77e3b86cb81de0ec0a8ce2a450ca0c170130d673)